libevmasm/PeepholeOptimiser.cpp

```diff
@@ -487,6 +498,47 @@ struct UnreachableCode
 	}
 };
 
+struct DeduplicateNextTag
+{
+	static bool apply(OptimiserState& _state)
+	{
+		static size_t maxBlockSize = 3;
+		for (size_t blockSize = 1; blockSize <= maxBlockSize; ++blockSize)
```

Comment:
I'd also expect that this would look much simpler if you'd split it up into several ``applySimple`` methods with a fixed block size each - especially since I think you said not all of them are actually needed anyways :-)

Reply:
Yeah, at first I thought that we would even want to test and possible use other sizes, so I made it generic.
I will make it simpler.